src/ProjectManagement/Utility/BuildIntegratedProjectUtility.cs

```diff
@@ -79,5 +77,77 @@ public static ExternalProjectReference ConvertProjectReference(BuildIntegratedPr
         {
             return new ExternalProjectReference(reference.Name, reference.PackageSpecPath, reference.ExternalProjectReferences);
         }
+
+        public static IReadOnlyList<PackageIdentity> GetOrderedProjectDependencies(BuildIntegratedNuGetProject buildIntegratedProject)
```

Comment:
We find the dependencies and order them at solution open. This isn't perfect since dependency order is per target framework, and at the solution level there could be multiple target frameworks for the same package if it is installed into multiple projects, but we determine the best order we can and then run them.